src/core/email-client.js

```diff
@@ -53,101 +71,250 @@ function getFreshImap (cb) {
+                self._email = email;
+                self._transporter.resolve(NodeMailer.createTransport(smtpConfig));
+                
             }
         });
     }
 
-    _checkEmails () {
+    _loadInfo(){
+        //  Get Imap obj and check emails
+        var self = this;
+        return self._imap.promise;
+    }
+
+    _loadTransporter(){
+        var self = this; 
+        return self._transporter.promise;
+    }
+
+    _loadStatus(){
+        var self = this;
+        return self._status.promise;
+    }
+
+    _loadSendCheck(){
+        var self = this;
+        return self._SendCheck.promise;
+    }
+
+    _loadChangeCheck(){
+        var self = this;
+        return self._changeCheck.promise;
+    }
+
+    //method to change name of user
+    _changeName(name){
+        var self = this;
+        self._name = name;
+    }
+
+    _changeEmail(email, password){
         var self = this;
+        //self._email = email;
+
+        self._imap = Q.defer();
+        self._transporter = Q.defer();
+
+        var obj = {
+          "email": email,
+          "password": password
+        };
+
+        jsonfile.writeFile("./email.json", obj, function (err) {
+            if (err){
+                console.log(err);
+                return;
+            }
+            console.log("email.json was updated!");
 
-        self.imap.on('ready', function () {
-            self.imap.openBox('INBOX', false, function (err, box) {
+            getFreshImap(function (err, imap, email, smtpConfig) {
                 if (err) {
-                    throw err;
+                    console.log(err);
+                } else {
+                    self._imap.resolve(imap);
+                    self._email = email;
+                    self._changeCheck.resolve(true);
+                    self._transporter.resolve(NodeMailer.createTransport(smtpConfig));
+                    
                 }
+            });            
 
-                //  Parses all unseen emails and marks them as read
-                var collectUnseenEmails = function () {
-                    self.imap.search(['UNSEEN'], function (err, results) {
-                        if (err || results.length === 0) {
-                            console.log(err || 'NO NEW MAIL');
-                            return;
-                        }
-
-                        //  Fetch the entire email bodies
-                        var f = self.imap.fetch(results, {
-                            bodies: [''],
-                            markSeen: true
-                        });
+        });
+
+
+    }
+
+    //attachments as array of Path
+    _sendEmail(toName, toEmail, subject, text, attachments){
+        var self = this;
+        var fromText = {
+            name: self._name,
+            address: self._email
+        };
+        var toText = {
+            name: toName, 
+            address: toEmail
+        };
+
+        var files = [];
+        if (attachments !== null){
```

Comment:
My above comment is incorrect. What you should do here is
`if (attachments) {}`
The main reason is in JS, variables can not only be null but also values like undefined. So if 'attachments' was set to undefined, it would still pass this check.